_posts/2024-04-01-review-topic-sentiment-overview.md

```diff
@@ -0,0 +1,56 @@
+
+## Challenges 
+
+Using LLMs to extract review topics and sentiments also has some challenges. A notable hurdle is the model's inclination to "correct" the UGC text, altering verbatim snippets in ways that makes it harder for us to detect the snippet in the original review. Additionally, navigating OpenAI's content filtering poses its own set of challenges. At times, certain reviews are filtered out for reasons that are not immediately apparent. While there might be some underlying sexual connotations in some reviews that are filtered, with others it is entirely weird why this input should be filtered out. For example, the following review does not pass OpenAI’s content filter: 
+
+*“Sehr schönes Hotel mit freundlichen Personal, sehr Hilfsbereit und sehr freundlich und sehr Sauber schöner Pool. Sehr schön Zimmer.”* (translation: *Very nice hotel with friendly staff, very helpful and very friendly and very clean, beautiful pool. Very nice room.*) 
+
+When it comes to our choice for which language models we use, there are a number of criteria at stake that need to be considered. Our choice in using GPT 3.5 was determined by a trade-off between quality of results (esp. with German language extraction) and budget. At the same time, we're exploring different avenues to use (fine-tuned) open source models as a replacement. Our experience in this respect has mostly shown two challenges: First, we still lack the same quality in the results for sentiment and topic extraction with the German language when using OSS models. Second, we also experience difficulties in achieving the same level of reliability for generating structured output in JSON format. Nevertheless, we are very confident that we will soon overcome these obstacles and use self-hosted models for these purposes. 
+
+These challenges underscore the complexities of leveraging advanced AI technologies in a way that respects the authenticity and diversity of user experiences. 
+
+## Evaluation 
+
+We invite you to read our tech blog post ([https://techblog.holidaycheck.com/post/2023/08/24/sentiment-analysis-llms](https://techblog.holidaycheck.com/post/2023/08/24/sentiment-analysis-llms)) where we outline how we test our sentiment analysis approaches. Within our teams, manual checks are a cornerstone of our evaluation process, providing a hands-on approach to refining and enhancing our models. We are committed to continuous improvement, relying on these meticulous reviews to polish our algorithms. Moreover, the feature's introduction to our users follows a strategic, gradual roll-out. This phased approach allows us to monitor the quality closely, make adjustments, and ensure that the insights we provide are both accurate and representative of the original review. 
```

Comment:
We invite you to read our tech blog post ([https://techblog.holidaycheck.com/post/2023/08/24/sentiment-analysis-llms](https://techblog.holidaycheck.com/post/2023/08/24/sentiment-analysis-llms)) where we outline how we test our sentiment analysis approaches. Manual checks (even though they might sound dull) are still a cornerstone of our teams' evaluation process, providing a hands-on approach to refining and enhancing our models. We are committed to continuous improvement, relying on these meticulous reviews to polish our algorithms. Moreover, the feature's introduction to our users follows a strategic, gradual roll-out. This phased approach allows us to monitor the quality closely, make adjustments, and ensure that our insights are accurate and representative of the original review.